packages/cli-kit/package.json

```diff
@@ -100,6 +100,7 @@
     "@oclif/core": "2.1.4",
     "@types/archiver": "5.3.2",
     "abort-controller": "3.0.0",
+    "ajv": "8.12.0",
```

Comment:
@gonzaloriestra I've tried to follow the history in the repository, and I think you've added the `zod` library, so I'm assuming you have the context, still, please, feel free to forward this message if you don't.

I'm adding `ajv` because it handles JSON schemas by loading them, while `zod` supports its own DSL. 

In themes, we really want to load the schemas from the JSON schema file because we're working on introducing schemas in the [shopify/theme-liquid-docs](https://github.com/shopify/theme-liquid-docs) in the middle-term. That's powerful because we keep [shopify/theme-liquid-docs](https://github.com/shopify/theme-liquid-docs) auto synced with some internal modules, so developers can get the most updated assets without a new tooling release (we're already doing that for Intelligent Code Completion).

`zod` does have an extension to load JSON schema files, however it doesn't support [data types definitions](https://github.com/Shopify/cli/pull/1909/files#diff-4a78aede52f87ba37d16049f0ca3160c3cd9b0178a61d1d68cafd59f29f49efeR44-R53). With that in mind, I feel that `ajv` is the way to go.

Do you have any questions/comments/concerns about it?

Thank you for the early review :)

Reply:
Thanks for the detailed explanation! I'm fine with adding that library if zod doesn't cover the functionality you need. 

My only concern is that it looks we will basically have two libraries for the same thing. So it would be great to be able to use only ajv for everything (which it also seems to be faster than zod, btw). The problem is that we are exporting zod as it is from cli-kit and using it extensively, so it will require some work and also breaking changes in cli-kit...

What do you think @Shopify/cli-foundations?

Reply:
In case we need to add the new library, I would retain both of them during a period of time. We could reimplement public libraries that uses `zod` with the new library. After some time we could deprecate the `zod` based APIs
IMHO I would try to avoid adding breaking changes without giving sometime to user to adopt the new APIs

Reply:
i'd avoid adding a new library if possible. zod has been great so far and replacing it would be a lot of work.
I just tested a "Json schema to zod" library, it does seems to work properly, what issues did you find @karreiro?
<img width="1270" alt="Captura de pantalla 2023-05-08 a las 13 16 04" src="https://user-images.githubusercontent.com/3757193/236810319-86f8fa05-8443-4939-bddc-9f69aba72bf9.png">

Reply:
FWIW you can also convert zod to JSON schema with https://www.npmjs.com/package/zod-to-json-schema

I'm using it in the demos and it works great!

The big downside is that zod is fairly legible and more TypeScript-like compared with JSON schema which is pretty wordy and doesn't make it as easy to do code completion. I think zod also supports more complex types.